R/hash.R

```diff
@@ -1,12 +1,29 @@
-
-# These return raw vectors of hashes. Vector elements are coded with
-# 32 bit hashes. Thus, the size of the raw vector of hashes is 4 times
-# the size of the input.
-
-vec_hash <- function(x) {
-  .Call(vctrs_hash, x)
+#' Hash a vector
+#'
+#' `vec_hash()` hashes a vector and returns the hash as a raw vector. Each
+#' element of `x` is coded with 32-bit hashes. This results in a raw vector
+#' that is 4 times the size of `x`.
+#'
+#' @param x A vector
+#' @param pool A logical. Should R's global string pool be used to hash
```

Comment:
It might make sense to use serialisation as in digest when `stable = TRUE`, since the serialisation format is versioned and stable.

Reply:
Really I'm uneasy about declaring our format stable though. Digest uses serialisation + well defined hashing algorithms to achieve stability. It feels wrong to commit to stability because our hashing routines are pieced together from disparate sources, we might want to change these pieces in the future.

Reply:
I really don't think we should claim this is stable, and I don't see the problem with mentioning R internals when they are stable and germane to the problem at hand.

Reply:
They may be stable but they are not set in stone. Other R implementations would prefer it if R gave up pointer equality for determining identity of objects, it's not entirely impossible that this changes in the future.

More importantly it seems like this option is trying to make hashing safe, or in any case more useful, for a particular kind of usage. Finding a word correctly describing that usage might help avoiding improper use.